syft/pkg/cataloger/rust/parse_audit_binary.go

```diff
@@ -61,3 +65,64 @@ func parseAuditBinaryEntry(reader unionreader.UnionReader, filename string) ([]r
 
 	return versionInfos, nil
 }
+
+// auditPkgPair is a helper struct to track the original index of the package in the original audit report + the syft package created for it
+type auditPkgPair struct {
+	pkg     *pkg.Package
+	rustPkg rustaudit.Package
+	index   int
+}
+
+func processAuditVersionInfo(location file.Location, versionInfo rustaudit.VersionInfo) ([]pkg.Package, []artifact.Relationship) {
```

Comment:
I would have loved to have used the `dependency.Resolve` but this requires material that is not persisted to a package, thus, cannot be used.

Reply:
I didn't quite understand why this is true. Can you explain what information is missing, and maybe add this as a comment in the file so future people working in here don't try to factor into dependency.Resolve?

Reply:
dependency.Resolve requires a Specifier function, which takes only a package. We tend to take information from the pkg.Metadata, however, in this case, the `rustaudit.VersionInfo` data structure is used to determine what the dependencies are which is not in the package metadata. I could add dependency indexes to the metadata, but that would be confusing since it is only useful if you have the original `[]rustaudit.Package` list (since they are indexes and not names). I probably shouldn't try and craft the name/name-version/name-version-source strings to do this lookup since that's not how they are referenced in the underlying material.